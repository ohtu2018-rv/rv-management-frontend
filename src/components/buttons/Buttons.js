import React from 'react';
import './styles/Buttons.css';

export const BasicBtn = ({ onClick, children, small, ...props }) => {
    const className = 'btn' + (small ? ' btn-small' : '');

    return (
        <button {...props} onClick={onClick} className={className}>
            {children}
        </button>
    );
};

export const SuccessBtn = ({ onClick, children, small, ...props }) => {
    const className = 'btn btn-success' + (small ? ' btn-small' : '');

    return (
        <button {...props} onClick={onClick} className={className}>
            {children}
        </button>
    );
};

export const DangerBtn = ({ onClick, children, small, ...props }) => {
    const className = 'btn btn-danger' + (small ? ' btn-small' : '');

    return (
        <button {...props} onClick={onClick} className={className}>
            {children}
        </button>
    );
};