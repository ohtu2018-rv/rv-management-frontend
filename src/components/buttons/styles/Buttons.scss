@import "vars";

.btn {
  border: none;
  border-radius: 3px;
  font-size: 16px;
  padding: 11px 16px;
  font-family: Arial, Helvetica, sans-serif;
  cursor: pointer;
  color: #000;
  -webkit-appearance: none;
  background: linear-gradient(to bottom, rgb(200, 200, 200), rgb(170, 170, 170));
}

.btn:active {
  box-shadow: inset 0 2px 3px 1px rgba(0, 0, 0, 0.15);
}

.btn-small {
  font-size: 14px;
  padding: 8px 14px;
}

.btn-success {
  color: #fff;
  background: linear-gradient(to bottom, rgb(93, 211, 63), rgb(77, 165, 37));
}

.btn-danger {
  color: #fff;
  background: linear-gradient(to bottom, rgb(192, 8, 8), rgb(168, 8, 8));
}
