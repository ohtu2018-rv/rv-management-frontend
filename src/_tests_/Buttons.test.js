import React from 'react';
import { BasicBtn, SuccessBtn, DangerBtn } from '../components/buttons/Buttons';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

it('renders buttons without crashing', () => {
    mount(<BasicBtn />);
    mount(<SuccessBtn />);
    mount(<DangerBtn />);

    // Snapshot tests
    const basicBtnTree = renderer.create(<BasicBtn />).toJSON();
    expect(basicBtnTree).toMatchSnapshot();
    const successBtnTree = renderer.create(<SuccessBtn />).toJSON();
    expect(successBtnTree).toMatchSnapshot();
    const dangerBtnTree = renderer.create(<DangerBtn />).toJSON();
    expect(dangerBtnTree).toMatchSnapshot();
});
