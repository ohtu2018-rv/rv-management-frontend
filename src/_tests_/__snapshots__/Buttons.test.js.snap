// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders buttons without crashing 1`] = `
<button
  className="btn"
  onClick={undefined}
/>
`;

exports[`renders buttons without crashing 2`] = `
<button
  className="btn btn-success"
  onClick={undefined}
/>
`;

exports[`renders buttons without crashing 3`] = `
<button
  className="btn btn-danger"
  onClick={undefined}
/>
`;
